datastore/google/cloud/datastore/_gax.py

```diff
@@ -90,20 +91,20 @@ class _DatastoreAPIOverGRPC(object):
     :type connection: :class:`Connection`
     :param connection: A connection object that contains helpful
                        information for making requests.
-
-    :type secure: bool
-    :param secure: Flag indicating if a secure stub connection is needed.
     """
 
-    def __init__(self, connection, secure):
-        if secure:
+    def __init__(self, connection):
+        parse_result = six.moves.urllib_parse.urlparse(
+            connection.api_base_url)
+        host = parse_result.hostname
```

Comment:
But if you look at [where][1] it is detected, I could easily set:

```
try:
    host = os.environ[GCD_HOST]
    self._base_url = 'http://' + host
    self._secure = False
except KeyError:
    self._base_url = _DATASTORE_BASE_URL
    self._secure = True
```

I just thought it might be overkill to carry around **two** values.

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/8b39bb002cf79e6805eae71e2957f950bba8225f/datastore/google/cloud/datastore/client.py#L208-L212